Formula/m/mihomo.rb

```diff
@@ -0,0 +1,53 @@
+class Mihomo < Formula
+  desc "Another rule-based tunnel in Go, formerly known as ClashMeta"
+  homepage "https://wiki.metacubex.one"
+  url "https://github.com/MetaCubeX/mihomo/archive/refs/tags/v1.18.5.tar.gz"
+  sha256 "205f6695ad19e232227c7e31da157860da9b2c4678900af9ebaf7e46c00c68c0"
+  license "GPL-3.0-or-later"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w -buildid=
+      -X "github.com/metacubex/mihomo/constant.Version=#{version}"
+      -X "github.com/metacubex/mihomo/constant.BuildTime=#{time.iso8601}"
+    ]
+    system "go", "build", "-tags", "with_gvisor", *std_go_args(ldflags:)
+  end
+
+  def post_install
+    (etc/"mihomo").mkpath
+    (etc/"mihomo/config.yaml").write <<~EOS
+      # Document: https://wiki.metacubex.one/config/
+      mixed-port: 7890
+    EOS
+  end
+
+  def caveats
+    <<~EOS
+      You need to customize #{etc}/mihomo/config.yaml.
+
+      mihomo can run without sudo if you don't need to enable TUN.
+
+      Can be used with Proxifier to replace TUN.
+    EOS
+  end
+
+  service do
+    run [opt_bin/"mihomo", "-d", etc/"mihomo"]
+    keep_alive true
+    working_dir etc/"mihomo"
+    log_path var/"log/mihomo.log"
+    error_log_path var/"log/mihomo.log"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/mihomo -v")
+
+    (testpath/"mihomo/config.yaml").write <<~EOS
+      mixed-port: 7890
```

Comment:
```suggestion
      mixed-port: #{free_port}
```

Reply:
Thank you very much. I have modified it according to your suggestion